lib/ship.rb

```diff
@@ -1,6 +1,16 @@
 class Ship
-  attr_reader :type
+  attr_reader :type, :length, :health
   def initialize(type, length)
     @type = type
+    @length = length
+    @health = length
+  end
+
+  def sunk?
+    health < 1
+  end
+
+  def hit
+    @health -= 1
```

Comment:
@RyanDBarnett 

Should we limit this so that it can't have a negative value? 

I think maybe this is taken care of in a later class where we specify that you can't hit a ship twice in the same coordinate.

Reply:
Yes I think this could be solved in a later class. I think the cell class has a fired_upon? boolean that we will use to prevent the same coordinate being hit twice. Great thinking about this though!